src/commands/branch.rs

```diff
@@ -231,18 +232,37 @@ fn cmd_branch_forget(
     command: &CommandHelper,
     args: &BranchForgetArgs,
 ) -> Result<(), CommandError> {
-    fn find_globs(view: &View, globs: &[String]) -> Result<Vec<String>, CommandError> {
-        let globs: Vec<glob::Pattern> = globs
-            .iter()
-            .map(|glob| glob::Pattern::new(glob))
-            .try_collect()?;
-        let matching_branches = view
-            .branches()
-            .iter()
-            .map(|(branch_name, _branch_target)| branch_name)
-            .filter(|branch_name| globs.iter().any(|glob| glob.matches(branch_name)))
-            .cloned()
-            .collect();
+    fn find_globs(view: &View, globs: &[String]) -> Result<IndexSet<String>, CommandError> {
+        let mut matching_branches: IndexSet<String> = IndexSet::default();
```

Comment:
No need to use `IndexSet`?

Reply:
What would you prefer? HashSet? BTreeSet?

I would use IndexSet everywhere that a set is iterated over, but I wasn't brave enough to add it as a dependency to the `lib` crate.

OTOH, it barely matters here (and in most cases). I also like IndexSet because it preserves order, which can be user-friendly, but that doesn't matter here either.

Reply:
`Vec`.

It could be `BTreeSet` as the caller builds it anyway.

We can also make it generic (something like `B: Default + Extend<String>`), but I don't think we'll need such complexity for private helper functions.